src/relentless/optimize/objective.py

```diff
@@ -375,6 +384,65 @@ def compute(self, variables, directory=None):
         return result
 
     def compute_gradient(self, ensemble, variables):
+        """Computes the relative entropy gradient for an ensemble.
+
+        Parameters
+        ----------
+        ensemble : :class:`~relentless.ensemble.Ensemble`
+            The ensemble for which to evaluate the gradient.
+        variables : :class:`~relentless.variable.Variable` or tuple
+            Variables with respect to which to compute gradient.
+
+        Returns
+        -------
+        :class:`~relentless.math.KeyedArray`
+            The gradient, keyed on the ``variables``.
+
+        """
+        if isinstance(self.target, Ensemble) and isinstance(
+            self.thermo, EnsembleAverage
+        ):
+            return self._compute_gradient_old(ensemble, variables)
+        else:
+            return self._compute_gradient_new(ensemble, variables)
+
+    def _compute_gradient_new(self, ensemble, variables):
+        r"""Computes the relative entropy gradient for an ensemble.
+
+        Parameters
+        ----------
+        ensemble : :class:`string`
+            Path to gsd trajectory file.
```

Comment:
We talked about this, but I just wanted to confirm we're good to just require this use GSD since it should contain all the type information. I don't think that is a problem since we interconvert for users.

Reply:
Yes, I think it's OK to say (at least for now) that it has to be a GSD file.